File: 백준/Gold/1931. 회의실 배정/회의실 배정.js
const rl = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  const n = input[0];
  let list = input.slice(1).map((v) => v.split(" ").map((v) => +v));

  const sorted = list.sort((a, b) => a[1] - b[1] || a[0] - b[0]);
  let count = 0;
  let lastEnd = 0;
  sorted.forEach(([s, e]) => {
    if (s >= lastEnd) {
      lastEnd = e;
      count++;
    }
  });

  console.log(count);
});

File: 백준/Gold/1931. 회의실 배정/회의실 배정.py
import sys

input = sys.stdin.readline
n = int(input())
meets = [tuple(map(int, input().split())) for _ in range(n)]

meets.sort(key=lambda x: (x[1], x[0]))


last_end = 0
cnt = 0
for s, e in meets:
    if s >= last_end:
        cnt += 1
        last_end = e

print(cnt)
